packages/react/ds/src/forms/form-field/form-field.tsx

```diff
@@ -0,0 +1,24 @@
+import { FormFieldLegacy } from './form-field-legacy/form-field-legacy.js';
+import { FormFieldNext } from './form-field-next/form-field-next.js';
+import { FormFieldProps } from './types.js';
+
+const deprecatedKeys = ['error', 'hint', 'label'] as const;
+
+const FormField = (props: FormFieldProps) => {
+  const isLegacy = deprecatedKeys.some((key) => key in props);
+
+  if (isLegacy) {
+    console.warn(
+      '[FormField] Using legacy props. Please migrate to the new composable API.',
+    );
+    return <FormFieldLegacy {...props} />;
```

Comment:
Instead of using `FormFieldLegacy`, can't we compose the `FormFieldNext` using the values passed with the old props?
Something like this:
```
if (isLegacy) {
    console.warn(
      '[FormField] Using legacy props. Please migrate to the new composable API.',
    );
    return (
        <FormFieldNext>
            {props.label && <FormField.Label>props.label</FormField.Label>}
            {props.hint && <FormField.Hint>props.hint</FormField.Hint>}
            {props.error && <FormField.Error>props.error</FormField.Error>}
            {props.children}
        </FormFieldNext>
    );
}
```